craft_cli/dispatcher.py

```diff
@@ -525,7 +527,11 @@ def pre_parse_args(
             if self._default_command is None:
                 help_text = self._get_general_help(detailed=False)
                 raise ArgumentParsingError(help_text)
-            emit.trace(f"Using default command: {self._default_command.name!r}")
+            emit.progress(
+                f"Using {self._app_name} without a command will be removed in a future release. "
+                f"Use '{self._app_name} {self._default_command.name}' instead.",
+                permanent=True,
+            )
```

Comment:
I used [this](https://github.com/canonical/snapcraft/pull/5524/files#diff-21dad9464134ee8702cb6b992710396334893b8788d5a2c0047d29aab2053011) and [this](https://github.com/canonical/snapcraft/pull/5523/files#diff-e1932468da3926078a95b4ef84e2e3daed2551b87b6d503a9e9caea3e84c109a) as reference for the wording.